src/processor/mine.rs

```diff
@@ -192,6 +196,26 @@ pub fn process_mine<'a, 'info>(
     Ok(())
 }
 
+pub fn vest_rewards(clock: &Clock, proof: &mut Proof) {
+    let hours_since_vest = clock
+        .unix_timestamp
+        .saturating_sub(proof.last_vest_at)
+        .saturating_div(ONE_HOUR);
+
+    for _ in 0..hours_since_vest.min(24) {
+        for i in (0..24).rev() {
+            proof.vesting[i + 1] = proof.vesting[i + 1].saturating_add(proof.vesting[i]);
```

Comment:
nit: `saturating_add` is only needed for the first iteration right? 
Not sure how much CU would this saved, though.